hadoop-common-project/hadoop-common/src/main/bin/hadoop-functions.sh

```diff
@@ -618,8 +618,7 @@ function hadoop_bootstrap
 
   export HADOOP_OS_TYPE=${HADOOP_OS_TYPE:-$(uname -s)}
 
-  # defaults
-  export HADOOP_OPTS=${HADOOP_OPTS:-"-Djava.net.preferIPv4Stack=true"}
+ export HADOOP_OPTS="${HADOOP_OPTS:-"-Djava.net.preferIPv4Stack=true"} -XX:+IgnoreUnrecognizedVMOptions --add-exports=java.base/sun.security.x509=ALL-UNNAMED --add-exports=java.base/sun.security.util=ALL-UNNAMED"
```

Comment:
I would suggest adding the JPMS opts in `function hadoop_finalize_hadoop_opts` or add a dedicated function `hadoop_finalize_jpms_opts`

also, we should add comments on both sides(here and pom.xml), to mention the future contributors sync the jpms opts

Reply:
I concur. we also need to make sure all forked processes (terasort...) are happy